pkg/controller/s3/bucket/subresources.go

```diff
@@ -29,22 +31,23 @@ type SubresourceClient interface {
 	CreateOrUpdate(ctx context.Context, bucket *v1beta1.Bucket) error
 	Delete(ctx context.Context, bucket *v1beta1.Bucket) error
 	LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error
+	SubresourceExists(bucket *v1beta1.Bucket) bool
 }
 
 // NewSubresourceClients creates the array of all clients for a given BucketProvider
-func NewSubresourceClients(client s3.BucketClient) []SubresourceClient {
+func NewSubresourceClients(client s3.BucketClient, l logging.Logger) []SubresourceClient {
```

Comment:
I'm not sure we want to log every late init call, even if it's debug but I'm fine either way.

Reply:
I think it was just useful to add this in so it is available to add more detailed logging in the future as well. This way when debugging these resources going forward, logging statements can be added in easily.